### PR TITLE
Aws private api deployment

### PR DESCRIPTION
**Pull request by Jeandsmith**
## What I am changing
Added an example for a deployment of a AWS private rest-api based on public-sector use developers. 

## How you can test it
No tests, simply deploy after setting the `vpc_endpoint_id` config value.

### CODE REVIEW

**Comment by vincentsarago** [comment]
that's interesting! from my understanding, the lambda is not directly within the VPC but only the API gateway?

**Comment by Jeandsmith** [comment]
The vpc endpoint security group can be configured to allow inbound requests from the Lambda function

File: deployment/aws/cdk/read.14868.1.lock [hunk 1]
```diff
@@ -0,0 +1 @@
+14868
```
Comment (vincentsarago) [inline deployment/aws/cdk/read.14868.1.lock:1]: I'm not sure what is this file and why it has to be included

File: deployment/aws/cdk/config.py [hunk 2]
```diff
@@ -85,4 +85,7 @@ class StackSettings(BaseSettings):
     # Default: - No specific limit - account limit.
     max_concurrent: Optional[int] = None
 
+    # The VPC Endpoint ID
+    vpc_endpoint_id: str = "{{VPCE_ID}}"
```
Comment (vincentsarago) [inline deployment/aws/cdk/config.py:89]: This should be optional (default to None), we can then add a check within the `TitilerPrivateApiStack` construct

File: deployment/aws/cdk/read.14868.1.lock (hunk 1 above)
Comment (Jeandsmith) [reply deployment/aws/cdk/read.14868.1.lock]: Removed. Not sure what was it either.

File: deployment/aws/cdk/config.py (hunk 2 above)
Comment (Jeandsmith) [reply deployment/aws/cdk/config.py]: Changed the config to optional and adapted the construct to it. Should be good.

**Pull request closed without merging**